File: sdks/python/src/boxsdk/models/operations/post_storage_policy_assignments.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils
from ..shared import clienterror as shared_clienterror
from ..shared import storagepolicyassignment as shared_storagepolicyassignment

class PostStoragePolicyAssignmentsRequestBodyAssignedToTypeEnum(str, Enum):
    USER = "user"
    ENTERPRISE = "enterprise"


@dataclass_json
@dataclasses.dataclass
class PostStoragePolicyAssignmentsRequestBodyAssignedTo:
    r"""PostStoragePolicyAssignmentsRequestBodyAssignedTo
    The user or enterprise to assign the storage
    policy to.
    """
    
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    type: PostStoragePolicyAssignmentsRequestBodyAssignedToTypeEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    
class PostStoragePolicyAssignmentsRequestBodyStoragePolicyTypeEnum(str, Enum):
    STORAGE_POLICY = "storage_policy"


@dataclass_json
@dataclasses.dataclass
class PostStoragePolicyAssignmentsRequestBodyStoragePolicy:
    r"""PostStoragePolicyAssignmentsRequestBodyStoragePolicy
    The storage policy to assign to the user or
    enterprise
    """
    
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    type: PostStoragePolicyAssignmentsRequestBodyStoragePolicyTypeEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    

@dataclass_json
@dataclasses.dataclass
class PostStoragePolicyAssignmentsRequestBody:
    assigned_to: PostStoragePolicyAssignmentsRequestBodyAssignedTo = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('assigned_to') }})
    storage_policy: PostStoragePolicyAssignmentsRequestBodyStoragePolicy = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('storage_policy') }})
    

@dataclasses.dataclass
class PostStoragePolicyAssignmentsRequest:
    request: Optional[PostStoragePolicyAssignmentsRequestBody] = dataclasses.field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    retries: Optional[utils.RetryConfig] = dataclasses.field(default=None)
    

@dataclasses.dataclass
class PostStoragePolicyAssignmentsResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    client_error: Optional[shared_clienterror.ClientError] = dataclasses.field(default=None)
    storage_policy_assignment: Optional[shared_storagepolicyassignment.StoragePolicyAssignment] = dataclasses.field(default=None)
    

File: sdks/ts/src/sdk/models/operations/poststoragepolicyassignments.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";


export enum PostStoragePolicyAssignmentsRequestBodyAssignedToTypeEnum {
    User = "user",
    Enterprise = "enterprise"
}


// PostStoragePolicyAssignmentsRequestBodyAssignedTo
/** 
 * The user or enterprise to assign the storage
 * policy to.
**/
export class PostStoragePolicyAssignmentsRequestBodyAssignedTo extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type: PostStoragePolicyAssignmentsRequestBodyAssignedToTypeEnum;
}

export enum PostStoragePolicyAssignmentsRequestBodyStoragePolicyTypeEnum {
    StoragePolicy = "storage_policy"
}


// PostStoragePolicyAssignmentsRequestBodyStoragePolicy
/** 
 * The storage policy to assign to the user or
 * enterprise
**/
export class PostStoragePolicyAssignmentsRequestBodyStoragePolicy extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type: PostStoragePolicyAssignmentsRequestBodyStoragePolicyTypeEnum;
}


export class PostStoragePolicyAssignmentsRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=assigned_to" })
  assignedTo: PostStoragePolicyAssignmentsRequestBodyAssignedTo;

  @SpeakeasyMetadata({ data: "json, name=storage_policy" })
  storagePolicy: PostStoragePolicyAssignmentsRequestBodyStoragePolicy;
}


export class PostStoragePolicyAssignmentsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PostStoragePolicyAssignmentsRequestBody;
}


export class PostStoragePolicyAssignmentsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  storagePolicyAssignment?: shared.StoragePolicyAssignment;
}
